mhs/common/mhs_common/workflow/tests/test_common.py

```diff
@@ -78,3 +80,39 @@ def setUp(self):
         actual_url = common.CommonWorkflow._extract_endpoint_url(endpoint_details)
 
         self.assertEqual(expected_url, actual_url)
+
+    @async_test
+    async def test_lookup_asid_details(self):
+        self.mock_routing_reliability.get_end_point.return_value = test_utilities.awaitable({
+            MHS_END_POINT_KEY: [MHS_END_POINT_VALUE],
+            MHS_TO_ASID: MHS_TO_ASID_VALUE
+        })
+
+        url, to_asid = await self.workflow._lookup_to_asid_details(INTERACTION_DETAILS)
+
+        self.mock_routing_reliability.get_end_point.assert_called_with(SERVICE_ID)
+        self.assertEqual(url, MHS_END_POINT_VALUE)
+        self.assertEqual(to_asid, MHS_TO_ASID_VALUE[0])
+
+    @async_test
+    async def test_lookup_asid_details_error(self):
+        self.mock_routing_reliability.get_end_point.side_effect = Exception()
+
+        with self.assertRaises(Exception):
+            await self.workflow._lookup_to_asid_details(INTERACTION_DETAILS)
+
+    @async_test
+    async def test_extract_no_to_asid_returned(self):
+        endpoint_details = {MHS_TO_ASID: []}
+
+        with self.assertRaises(IndexError):
+            common.CommonWorkflow._extract_asid(endpoint_details)
+
+    @async_test
+    async def test_extract_asid_multiple_returned(self):
+        expected_asid = "asid 1"
+        endpoint_details = {MHS_TO_ASID: [expected_asid, "asid 2"]}
+
+        actual_asid = common.CommonWorkflow._extract_asid(endpoint_details)
+
+        self.assertEqual(expected_asid, actual_asid)
```

Comment:
Missing newline at the end of this file.

Reply:
It's still missing, isn't it?

Reply:
🤦‍♂